test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs

```diff
@@ -1232,5 +1628,81 @@ private class OverriddenValidBoundAttributeWithInvalidUpperCase : TagHelper
             [HtmlAttributeName("DATA-SOMETHING")]
             public string ValidSomething { get; set; }
         }
+
```

Comment:
Should add cases where `[HtmlAttributeName("valid-name", DictionaryAttributePrefix = null)]`